models/DataObject/Concrete.php

```diff
@@ -881,4 +881,62 @@ public static function getLatestVersionByObjectIdAndLatestModificationDate($obje
 
         return;
     }
+
+    /**
+     * @param array $descriptor
+     * @return array
+     */
+    public function retrieveRelationData($descriptor) {
+
+        if ($this instanceof CacheRawRelationDataInterface) {
+            $unfilteredData = $this->__getRawRelationData();
+
+            $likes = [];
+            foreach ($descriptor as $column => $expectedValue) {
+                if (is_string($expectedValue)) {
+                    $trimmed = rtrim($expectedValue, '%');
+                    if (strlen($trimmed) < strlen($expectedValue)) {
+                        $likes[$column] = $trimmed;
+                    }
+                }
+            }
+
+            $filterFn = function($row) use ($descriptor, $likes) {
```

Comment:
@weisswurstkanone 
for a micro-optimzation we can use `static function` here as `$this` context is not needed